lib/puppet/provider/logical_volume/lvm.rb

```diff
@@ -134,20 +134,19 @@ def size
     end
 
     def size=(new_size)
-        lvm_size_units = { "K" => 1, "M" => 1024, "G" => 1048576, "T" => 1073741824, "P" => 1099511627776, "E" => 1125899906842624 }
-        lvm_size_units_match = lvm_size_units.keys().join('|')
+        lvm_size_units = { "K" => 1, "M" => 1024, "G" => 1024**2, "T" => 1024**3, "P" => 1024**4, "E" => 1024**5 }
 
         resizeable = false
         current_size = size()
 
-        if current_size =~ /(\d+\.{0,1}\d{0,2})(#{lvm_size_units_match})/i
+        if current_size =~ /^([0-9]+(\.[0-9]+)?)([KMGTPE])/i
             current_size_bytes = $1.to_f
-            current_size_unit  = $2.upcase
+            current_size_unit  = $3.upcase
```

Comment:
Is it really an issue if all the lvm commands called are passed with a lowercase unit, or just as bytes?

I don't think the module should have to support uppercase meaning multiples of 1000 instead of 1024.  It should always assume 1024 as long as it calls the correct lvm commands (i.e. don't use uppercase units when calling lvm)